torch/csrc/jit/codegen/cuda/root_domain_map.cpp

```diff
@@ -76,6 +80,11 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseRootDomainMap::map(
   if (consumer_tv_->definition()->isA<TransposeOp>()) {
     return mapTranspose(
         producer, consumer, root_dims_to_map, producer_to_consumer);
+  } else if (
+      consumer_tv_->definition()->isA<TorchGatherOp>() &&
+      require_same_extent_) {
+    // Nothing to map when having same extent is required
+    return {};
   }
```

Comment:
Does this mean if `require_same_extent_` is true, then the consumer will not be mapped to the index tensor as well? I don't think this is correct.

Reply:
Ah, my bad. Will fix